ebpf/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/grafana/pyroscope/ebpf
 
-go 1.22
```

Comment:
For updating go in go.mod, you would need to:

Update it here too: https://github.com/grafana/pyroscope/blob/69fc34993d94791e2b7e85679ce76172e804f0a0/Makefile#L185-L190

and run `make go/mod`


And there is a script for the compile version: https://github.com/grafana/pyroscope/blob/main/tools/upgrade-go-version.sh (in case we want to upgrade it too)